spec/components/trainees/confirmation/course_details/view_spec.rb

```diff
@@ -24,42 +25,64 @@ module CourseDetails
             render_inline(View.new(data_model: trainee))
           end
 
-          it "tells the user that no data has been entered for course type, subject, age range, course start date and course end date" do
+          it "tells the user that no data has been entered for course details, course type, subject, age range, course start date and course end date" do
             found = component.find_all(".govuk-summary-list__row")
 
-            expect(found.size).to eq(5)
+            expect(found.size).to eq(6)
 
-            found.each do |row|
+            found.at(1..5).each do |row|
               expect(row.find(".govuk-summary-list__value")).to have_text(t("components.confirmation.not_provided"))
             end
           end
         end
 
         context "when data has been provided" do
-          let(:trainee) { build(:trainee, :with_course_details, id: 1) }
+          context "with a publish course" do
+            let(:trainee) { create(:trainee, :with_course_details, id: 1) }
+            let(:course) { instance_double("course", name: "some_name", code: "some_code") }
 
-          before do
-            render_inline(View.new(data_model: trainee))
-          end
+            before do
+              expect(Course).to receive(:find_by).with(code: trainee.course_code).and_return(course)
+              render_inline(View.new(data_model: trainee))
+            end
 
-          it "renders the course type" do
-            expect(component.find(".govuk-summary-list__row.type-of-course .govuk-summary-list__value"))
-              .to have_text(trainee.training_route.humanize)
-          end
+            it "renders the course details" do
+              expect(component.find(".govuk-summary-list__row.course-details .govuk-summary-list__value"))
```

Comment:
Could we put `component.find(".govuk-summary-list__row.course-details .govuk-summary-list__value"` into a `subject { component.find(".govuk-summary-list__row.course-details .govuk-summary-list__value" }` so we can assert on the subject? Then we won't need to duplicate the find across the other scenarios